devito/stencilkernel.py

```diff
@@ -116,12 +119,16 @@ def __init__(self, stencils, **kwargs):
         nodes = SubstituteExpression(subs=subs).visit(nodes)
 
         # Apply the Devito Loop Engine for loop optimization and finalize instantiation
-        dle_state = transform(nodes, mode=dle)
+        dle_state = transform(nodes, mode=set_dle_mode(dle, self.compiler),
+                              compiler=self.compiler)
         body = dle_state.nodes
```

Comment:
Yes, we should provide an explicit interface for external codes to retrieve and set the parameter order (we don't care about the ordering internally). I'll file a separate issue for this and make sure we address it before unleashing 3.0 to the wider public.